### PR TITLE
feat(notification): add philips hue

### PR DESCRIPTION
**Pull request by MattieX**
### Description
Added Philips Hue notification support over LAN or Cloud using Node-Hue-Api (See https://www.npmjs.com/package/node-hue-api).

Can set the color of the lights, whether they should blink, and individual light IDs to adjust.

<details>
<summary>Environment Variables</summary>

| Environment variable | Description | Notes |
|:---:|---|---|
| `PHLIPS_HUE_API_KEY` | Hue Bridge API Key | **Required**, generate key using instructions [here](https://developers.meethue.com/develop/get-started-2/). This will be used for both LAN and cloud access over the official Remote Hue API. |
| `PHLIPS_HUE_LAN_BRIDGE_IP` | LAN IP Address of your Hue Bridge | LAN only, e.g. `192.168.x.x`|
| `PHLIPS_HUE_LIGHT_IDS` | Light IDs | Optional (all if not supplied). Comma seperated, e.g.: `1`, `2` |See Hue App → About for IDs |
| `PHLIPS_HUE_LIGHT_COLOR` | Color in RGB Format | Optional (NVIDIA green if not supplied). Comma separated, e.g.: `255`, `255`, `255`|
| `PHLIPS_HUE_LIGHT_PATTERN` | `blink` or empty | Optional - lights will flash for 30 seconds if `blink` is supplied. |
| `PHLIPS_HUE_CLOUD_ACCESS_TOKEN` | Remote Access Token | Cloud only, the access token obtained from Philips's Remote Hue API. Instructions to generate [here](https://developers.meethue.com/develop/hue-api/remote-authentication/). |
| `PHLIPS_HUE_CLOUD_REFRESH_TOKEN` | Remote Refresh Token | Cloud only, the refresh token obtained from Philips's Remote Hue API.  |
| `PHLIPS_HUE_CLOUD_CLIENT_ID` | Remote Client ID | Cloud only, the client ID to use when accessing the Remote Hue API. |
| `PHLIPS_HUE_CLOUD_CLIENT_SECRET` | Remote Client Secret | Cloud only, the client secret to use when accessing the Remote Hue API.   |

</details>

### Testing
Used `npm run test:notification` after filling in required environment variables, observed the resulting change in Hue lights.
 

### New dependencies
[`node-hue-api`](https://www.npmjs.com/package/node-hue-api)

### CODE REVIEW

**Comment by jef** [comment]
LOL... This is pretty awesome. I'd love to see this come in. If you have any demo to video that would be actually amazing.

**Comment by banksio** [comment]
Checks should all be passing now, please let us know if we need to make any changes!

**Comment by MattieX** [comment]
> LOL... This is pretty awesome. I'd love to see this come in. If you have any demo to video that would be actually amazing.

Sure! [Here](https://www.youtube.com/watch?v=EI2fWgsLOkg) you are.

**Comment by shortywz** [comment]
> > LOL... This is pretty awesome. I'd love to see this come in. If you have any demo to video that would be actually amazing.
> 
> Sure! https://www.youtube.com/watch?v=EI2fWgsLOkg you are.

**This** is what open source is all about 😄

**Comment by jef** [comment]
> > LOL... This is pretty awesome. I'd love to see this come in. If you have any demo to video that would be actually amazing.
> 
> Sure! [youtube.com/watch?v=EI2fWgsLOkg](https://www.youtube.com/watch?v=EI2fWgsLOkg) you are.

Omg... I love it.

Thank YOU!

File: README.md [hunk 1]
```diff
@@ -409,8 +425,16 @@ environment variables are **optional**._
 
 </details>
 
+
+<details>
```
Comment (jef) [inline README.md:429]:
```suggestion
```

I don't think we want this one here.

File: src/notification/philips-hue.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+const refreshToken = hue.refreshToken;
+const remoteApiUsername = hue.remoteApiUsername;
+
+// Default Light State
+const lightState = new LightState()
+	.on(true)
+	.brightness(100)
+	.rgb(46.27, 72.55, 0);
+
+const adjustLightsWithAPI = (hueBridge: import('node-hue-api/lib/api/Api')) => {
```
Comment (jef) [inline src/notification/philips-hue.ts:24]: I'd prefer you import the type above and use here.

**Review by jef — CHANGES_REQUESTED** [review]
Overall, looks pretty good! Just a couple things and we can get this in!

**Review by jef — APPROVED** [review]
AWESOME! Thank you!

**Comment by jef** [comment]
@MattieX, do you mind if I link your video in the README?

**Comment by MattieX** [comment]
> @MattieX, do you mind if I link your video in the README?

Sure - I've updated it to include the new project name now. Here you are: https://vimeo.com/476083242

**Comment by jef** [comment]
That's thoughtful of you! Thanks for doing that. I'll add now!

**Pull request merged**